traad/app.py

```diff
@@ -368,18 +368,36 @@ def code_assist_calltip_view(context):
     }
 
 
-@app.get('/code_assist/definition')
+@app.post('/code_assist/definition')
 def code_assist_definition_view(context):
     args = bottle.request.json
 
     log.info('get definition: {}'.format(args))
 
-    return {
-        'results': context.workspace.get_definition_location(
-            code=args['code'],
-            offset=args['offset'],
-            path=args['path'])
-    }
+    with open(args['path'], 'r') as f:
+        code = f.read()
```

Comment:
Are you not going to pass the code along in the request? This could be problematic if users are looking at a modified buffer in emacs.

Reply:
Good point - I wondered why it was being sent directly beforehand rather than following the pattern of refactoring requests. This is a better approach.